sotodlib/site_pipeline/make_position_match.py

```diff
@@ -0,0 +1,883 @@
+
+    # Figure out the tuneset and check the list of obs
+    SMURF = ls.G3tSmurf(db_path=ctx["obsdb"], archive_path=None)
+    ses = SMURF.Session()
+    tunefile = []
+    for obs_id in obs_ids:
+        try:
+            obs = (
+                ses.query(ls.Observations)
+                .filter(ls.Observations.obs_id == obs_id)
+                .one()
+            )
+        except:
+            raise ValueError(
+                obs_id + " not found. Please check obs_id list and context file."
+            )
+        tunefile.append(obs.tunesets[0].path)
+    if not np.all(np.array(tunefile) == tunefile[0]):
+        raise ValueError("Not all observations have the same tunefile")
+    tunefile = tunefile[0]
+
+    # Build output path
+    ufm = config["ufm"]
+    append = ""
+    if "append" in config:
+        append = "_" + config["append"]
+    if len(obs_ids) == 1:
+        create_db(config["manifest_db"])
+        db = metadata.ManifestDb(config["manifest_db"])
+        outpath = os.path.join(config["outdir"], f"{ufm}_{obs.obs_id}{append}.h5")
+    else:
+        outpath = os.path.join(
+            config["outdir"], f"{ufm}_{obs.tunesets[0].id}{append}.h5"
+        )
+    dataset = "focal_plane"
+    input_paths = "input_data_paths"
+    outpath = os.path.abspath(outpath)
+
+    # Make ResultSet of input paths for later reference
+    types = ["config", "tunefile", "bgmap", "results", "context"]
+    paths = [
+        args.config_path,
+        tunefile,
+        config["bias_map"],
+        outpath,
+        config["context"]["path"],
+    ]
+    paths = [os.path.abspath(p) for p in paths]
+    types += ["obs_id"] * len(obs_ids)
+    paths += list(obs_ids)
+    rset_paths = metadata.ResultSet(
+        keys=["type", "path"],
+        src=np.vstack((types, paths)).T,
+    )
+
+    valid_bg = (0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11)
+
+    # If requested generate a template for the UFM with the instrument model
+    gen_template = config["gen_template"]
+    if gen_template:
+        logger.info("Generating template for " + ufm)
+        wafer = MapMaker(north_is_highband=False, array_name=ufm)
+        det_x = []
+        det_y = []
+        polang = []
+        det_ids = []
+        template_bg = []
+        is_north = []
+        for det in wafer.grab_metadata():
+            if not det.is_optical:
+                continue
+            det_x.append(det.det_x)
+            det_y.append(det.det_y)
+            polang.append(det.angle_actual_deg)
+            det_ids.append(det.detector_id)
+            template_bg.append(det.bias_line)
+            is_north.append(det.is_north)
+        det_ids = np.array(det_ids)
+        template_bg = np.array(template_bg)
+        template_msk = np.isin(template_bg, valid_bg)
+        template_n = np.array(is_north) & template_msk
+        template_s = ~np.array(is_north) & template_msk
+        template_bg[template_bg % 2 == 1] *= -1
```

Comment:
This `*= -1` occurs in several places.  I cannot figure out what the point of that is.

Reply:
Ah, that is a bit of a dumb trick. All it really does it punish the fitting algorithm for mixing up adjacent bias lines. Causes a moderate increase in my agreement with detmap. I could instead strictly enforce that the bias lines from `bgmap` but the current approach can handle some amount of error in `bgmap`. I actually think that the most "correct" thing to do is apply some non-affine transformation to the bias line information so it can guide the transformation in a way that cannot be reversed by the transformation itself. But that is something I want to think more deeply about before implementing.